main.c

```diff
@@ -0,0 +1,240 @@
+	{6.0, 0.0, 0.625},
+};
+static struct magnitude mags_y[] = {
+	{0.7, 0.0, 1.0},
+};
+
+
+static void resize(int w, int h)
+{
+	double ratio;
+
+	if (h == 0) {
+		h = 1;
+	}
+
+	ratio = (double)w/(double)h;
+
+	glMatrixMode(GL_PROJECTION);
+	glLoadIdentity();
+	glViewport(0, 0, w, h);
+	gluPerspective(FOV, ratio, 1, RENDER_DIST);
+	glMatrixMode(GL_MODELVIEW);
+}
+
+static void genNoise(double *tab, size_t len)
+{
+	size_t i;
+
+	for (i = 0; i < len; i++) {
+		tab[i] += (0.0005 * (rand() % 100));
+	}
+}
+
+static void *genWaveMap(size_t width, size_t length)
+{
+	size_t i;
+	size_t x, y;
+	size_t size;
+	double *map;
+	double *tmp;
+	int err;
+
+	size = width * length;
+	map = calloc(size, sizeof(double));
+	tmp = calloc(size, sizeof(double));
+
+	for (i = 0; i < size; i += width) {
+		err = idftCalc(ctx_x, &map[i], width);
+		if (err) {
+			exit(err);
+		}
+
+		err = idftCalc(ctx_y, &tmp[i], width);
+		if (err) {
+			exit(err);
+		}
+	}
+
+	for (y = 0; y < SIZE; y++) {
+		for (x = 0; x < SIZE; x++) {
+			map[y*SIZE + x] *= tmp[x*SIZE + y];
+		}
+	}
+
+	genNoise(map, size);
+
+	free(tmp);
+	return map;
+}
+
+static double *calcNormal(double *a, double *b, double *c)
+{
+	size_t i;
+	double ab[3];
+	double ac[3];
+	double len;
+	static double norm[3];
+
+	/* AB = (Bx - Ax, By - Ay, Bz - Az) */
+	for (i = 0; i < 3; i++) {
+		ab[i] = b[i] - a[i];
+		ac[i] = c[i] - a[i];
+	}
+
+	/* ABxAC = (a, b, c) */
+	norm[0] = ab[1]*ac[2] - ab[2]*ac[1];
+	norm[1] = ab[2]*ac[0] - ab[0]*ac[2];
+	norm[2] = ab[0]*ac[1] - ab[1]*ac[0];
+
+	len = sqrt(norm[0]*norm[0] + norm[1]*norm[1] + norm[2]*norm[2]);
+	for(i = 0; i < 3; i++) {
+		norm[i] /= len;
+	}
+
+	return norm;
+}
+
+static void render(void)
+{
+	size_t x, y;
+	double (*map)[SIZE][SIZE];
+	double a[3], b[3], c[3], d[3];
+	float wave_diff[] = {0.0f, 0.6f, 1.0f, 1.0f};
+	float wave_specular[] = {1.0f, 1.0f, 1.0f, 1.0f};
+	float wave_ambient[] = {0.0f, 0.1f, 0.2f, 1.0f};
+	float wave_shine[] = {100.0f};
+
+	glClear(GL_COLOR_BUFFER_BIT);
+
+	glMatrixMode(GL_MODELVIEW);
+	glLoadIdentity();
+	gluLookAt(CAMERA_X, CAMERA_Y, CAMERA_Z, EYE_X, EYE_Y, 0, 0, 0, 1);
+
+	map = genWaveMap(SIZE, SIZE);
+
+	glMaterialfv(GL_FRONT, GL_DIFFUSE, wave_diff);
+	glMaterialfv(GL_FRONT, GL_SPECULAR, wave_specular);
+	glMaterialfv(GL_FRONT, GL_AMBIENT, wave_ambient);
+	glMaterialfv(GL_FRONT, GL_SHININESS, wave_shine);
+
+	glBegin(GL_TRIANGLES);
+
+	for (x = 0; x < SIZE - 1; x++) {
+		for (y = 0; y < SIZE - 1; y++) {
+			a[0] = x;	a[1] = y;	a[2] = (*map)[x][y];
+			b[0] = x + 1;	b[1] = y;	b[2] = (*map)[x+1][y];
+			c[0] = x + 1;	c[1] = y + 1;	c[2] = (*map)[x+1][y+1];
+			d[0] = x; 	d[1] = y + 1;	d[2] = (*map)[x][y+1];
+
+			glNormal3dv(calcNormal(a, b, c));
+			glVertex3dv(a);
+			glVertex3dv(b);
+			glVertex3dv(c);
+
+			glNormal3dv(calcNormal(a, c, d));
+			glVertex3dv(a);
+			glVertex3dv(c);
+			glVertex3dv(d);
+		}
+	}
+
+	free(map);
+	glEnd();
+	glutSwapBuffers();
+}
+
+
+int main(int argc, char** argv)
+{
+	int err;
+
+	srand(1);
```

Comment:
What is the purpose of calling this function? 
As I understand if srand() is not called, the rand() seed is set as if srand(1) were called at the program start by default.

Reply:
I agree that standard dictates that RNG is initialized as srand(1) by default, but I wanted to make an explicit initialization so that in the future I can modify the seed.